react-router/react-router.d.ts

```diff
@@ -247,7 +247,7 @@ declare namespace ReactRouter {
       isActive: (pathOrLoc: H.LocationDescriptor, indexOnly?: boolean) => boolean
     }
 
-    function withRouter<C extends React.ComponentClass<any>>(component: C): C
+    function withRouter<C extends React.ComponentClass<any>>(component: C): C & InjectedRouter
```

Comment:
IMO this is wrong. InjectedRouter type describe a router property which is injected to a component wrapped with withRouter HOC. It rather should be something like:
```javascript
withRouter<C extends React.ComponentClass<TProps>, TProps extends { router?: InjectedRouter}>(component: C): C 
```